packages/server-wallet/src/protocols/depositing.ts

```diff
@@ -0,0 +1,50 @@
+import {right} from 'fp-ts/lib/Either';
+import {none, some} from 'fp-ts/lib/Option';
+import {BN} from '@statechannels/wallet-core';
+
+import {match} from '../match';
+import {Address, Uint256} from '../type-aliases';
+
+import {Protocol, ChannelState, ProtocolResult} from './state';
+
+export type ProtocolState = ChannelState & {
+  assetHolderAddress: Address;
+  amountToDeposit: Uint256;
+  safeToDepositAmount: Uint256;
+  fundedAt: Uint256;
+  depositTransactionId: string | undefined;
+};
```

Comment:
I'd rather if the `Depositing` protocol state looked like
```
{
  app: ChannelState,
  assetHolderAddress: Address;
  amountToDeposit: Uint256;
  safeToDepositAmount: Uint256;
  fundedAt: Uint256;
  depositTransactionId: string | undefined;
}
```

We have been burned a few times by defining flat data types like this, and then using the spread operator to create them.

Reply:
This might be a good pattern to follow across all our `ProtocolState`